### DISCUSSION TITLE
Duplicate data in React js

### DISCUSSION THREAD

**Discussion by bobwatcherx** [supabase/supabase#5238] [General]
i have 4 data in table but i Get 8 data in React js

output :
[{"id":1,"created_at":"2022-01-30T08:18:27+00:00","nama_pengirim":"jono"},{"id":2,"created_at":"2022-01-30T08:24:34+00:00","nama_pengirim":"bekasi"},{"id":3,"created_at":"2022-01-30T08:30:07.326568+00:00","nama_pengirim":"bego123"},{"id":4,"created_at":"2022-01-30T08:30:30.663116+00:00","nama_pengirim":"bego123"}]
[{"id":1,"created_at":"2022-01-30T08:18:27+00:00","nama_pengirim":"jono"},{"id":2,"created_at":"2022-01-30T08:24:34+00:00","nama_pengirim":"bekasi"},{"id":3,"created_at":"2022-01-30T08:30:07.326568+00:00","nama_pengirim":"bego123"},{"id":4,"created_at":"2022-01-30T08:30:30.663116+00:00","nama_pengirim":"bego123"}]

my code :

```js
import { createClient } from '@supabase/supabase-js'

const supabase = createClient('https://SECRET.supabase.co', 'secret')

 
export default function Historypay(){
	const [post,setPost] = useState([])
	async function fetchdata(){
            const {data} = await supabase
                            .from('tblpesanan')
                            .select()
                            
            setPost(data)
            console.log(data)
	}
	
      useEffect(async()=>{
	      fetchdata()
      },[])

	return(
		<div>
		{JSON.stringify(post)}
		</div>
        )
}
```

**Comment by kiwicopple** [comment]
hey @bobwatcherx - the code is looking pretty good!

Im not sure if this is a supabase bug - likely just a bug in the way you've set up your react component. I'll convert this to a Discussion

Is your code public? Perhaps you can share a link to the repo? It looks fine from a initial scan, so it could be something outside of the code you've shared here
